docs/running.rst

```diff
@@ -515,6 +515,105 @@ They are summarized below:
+            * ...
+        - PrgEnv2
+            : perf_variable1: <value> <units>
+            : perf_variable2: <value> <units>
+            * ...
+    ------------------------------------------------------------------------------
+    Check2
+    - system:partition
+        - PrgEnv1
+            * perf_variable1: <value> <units>
+            * perf_variable2: <value> <units>
+            * ...
+        - PrgEnv2
+            * perf_variable1: <value> <units>
+            * perf_variable2: <value> <units>
+            * ...
+    ------------------------------------------------------------------------------
+
+where the performance variables are listed by programming environment and partition for each of the tests that are run.
+
+For instance
+
+  .. code-block:: bash
+  reframe -C tutorial/config/settings.py -c tutorial/example7.py -r --performance-report
+
+prints something like this
+
+  .. code-block:: bash
+
+  Command line: ./reframe -r -c tutorial/example7.py --performance-report
+  Reframe version: 2.18-dev1
+  Launched by user: USER
+  Launched on host: daint101
+  Reframe paths
+  =============
+      Check prefix      : 
+      Check search path : 'example7.py'
+      Stage dir prefix     : /path/to/reframe/stage/
+      Output dir prefix    : /path/to/reframe/output/
+      Perf. logging prefix : /path/to/reframe/perflogs
+  [==========] Running 1 check(s)
+  [==========] Started on Mon Apr 15 13:49:34 2019 
+  
+  [----------] started processing Example7Test (Matrix-vector multiplication (CUDA performance test))
+  [ RUN      ] Example7Test on daint:gpu using PrgEnv-cray
+  [       OK ] Example7Test on daint:gpu using PrgEnv-cray
+  [ RUN      ] Example7Test on daint:gpu using PrgEnv-gnu
+  [       OK ] Example7Test on daint:gpu using PrgEnv-gnu
+  [ RUN      ] Example7Test on daint:gpu using PrgEnv-pgi
+  [       OK ] Example7Test on daint:gpu using PrgEnv-pgi
+  [----------] finished processing Example7Test (Matrix-vector multiplication (CUDA performance test))
+  
+  [  PASSED  ] Ran 3 test case(s) from 1 check(s) (0 failure(s))
+  [==========] Finished on Mon Apr 15 13:55:22 2019 
+  ==============================================================================
+  PERFORMANCE REPORT
+  ------------------------------------------------------------------------------
+  Example7Test
+  - daint:gpu
+     - PrgEnv-cray
+        * perf: 49.994311 Gflop/s
+     - PrgEnv-gnu
+        * perf: 50.748701 Gflop/s
+     - PrgEnv-pgi
+        * perf: 49.844147 Gflop/s
+  ------------------------------------------------------------------------------
+
+If a test needs to be run on a situation where there are no references, the `self.reference` dictionary may be written as
+
+  .. code-block:: bash
+   
+  self.reference = {
+      '*': {
+          'perf_variable1': (0, None, None),
+          'perf_variable2': (0, None, None)
+          ...
+      }
+  }
+
+This will pass the performance checks and the performance report will be printed.
```

Comment:
According to the above, the test is going to pass the performance test and the performance report is going to be printed as expected.